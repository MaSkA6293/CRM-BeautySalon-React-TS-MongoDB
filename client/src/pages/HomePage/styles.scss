.Home {
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  align-content: flex-start;
  border-radius: 0 10px 10px 0;
  min-height: 100vh;
  &__menu {
    width: 100%;
    min-width: 320px;
  }
  &__NavItem {
    min-width: 100%;
    padding: 5px;
    margin: 5px 0;
  }
  &__Label {
    padding: 10px 0;
  }
}
