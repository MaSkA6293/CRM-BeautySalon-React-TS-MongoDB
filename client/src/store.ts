import { createStore, compose, applyMiddleware } from "redux";
import rootReducer from "./reducers/rootReducer";
//import thunk from "redux-thunk";

import createSagaMiddleware from "redux-saga";
import rootSaga from "./sagas";

const sagaMiddleware = createSagaMiddleware();
const composeEnhancers =
  process.env.NODE_ENV !== "production" &&
    typeof window === "object" &&
    (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
    : compose;

const configureStore = () =>
  createStore(
    rootReducer,
  //  composeEnhancers(applyMiddleware(thunk, sagaMiddleware))
    composeEnhancers(applyMiddleware( sagaMiddleware))
  );

const store = configureStore();
sagaMiddleware.run(rootSaga);
export default store;
