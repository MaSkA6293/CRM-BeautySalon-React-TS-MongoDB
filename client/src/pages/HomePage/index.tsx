import React from "react";
import "./styles.scss";
import "react-datepicker/dist/react-datepicker.css";
import NavItem from "./components/NavItem";
import { faUsers, faList } from "@fortawesome/free-solid-svg-icons";
import { Link } from "react-router-dom";
import ListItem from "@material-ui/core/ListItem";
import HomeHeader from "./components/HomeHeader";
const Home = () => {
  return (
    <div className="Home">
      <HomeHeader />
      <div className="Home__menu">
        <Link to="/clients">
          <ListItem button>
            <NavItem
              icon={faUsers}
              title={"Клиенты"}
              addClass={["Home__NavItem"]}
            />
          </ListItem>
        </Link>
        <Link to="/services">
          <ListItem button>
            <NavItem
              icon={faList}
              title={"Услуги"}
              addClass={["Home__NavItem"]}
            />
          </ListItem>
        </Link>
      </div>
    </div>
  );
};

export default Home;
