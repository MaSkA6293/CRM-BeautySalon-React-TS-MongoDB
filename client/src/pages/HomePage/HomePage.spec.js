import React from "react";
import HomePage from "./index";
import { Provider } from "react-redux";
import store from "../../store";

import { BrowserRouter as Router } from "react-router-dom";
const setUp = (props) =>
    render(
        <Provider store={store}>
            <Router>
                <HomePage {...props} />
            </Router>
        </Provider>,
    );

describe("all tests HomePage", () => {
    let component;
    beforeEach(() => {
        component = setUp();
    });
    describe("HomePage component", () => {
        it("should render HomePage component", () => {
            expect(component).toMatchSnapshot();
        });
    });
});
