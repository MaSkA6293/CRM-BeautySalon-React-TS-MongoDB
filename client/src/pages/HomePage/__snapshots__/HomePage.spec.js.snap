// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all tests HomePage HomePage component should render HomePage component 1`] = `
<div
  class="home"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary header mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="home-header"
      >
        <span
          class="home-header__label"
        >
          Beauty Servise
        </span>
        <div
          class="home-header__menu"
        >
          <button
            aria-controls="simple-menu"
            aria-haspopup="true"
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            title="Настройки"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <div
                class="home-header__icon-menu"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
              </div>
            </span>
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    id="back-to-top-anchor"
  />
  <div
    class="makeStyles-root-1"
    role="presentation"
    style="transform:scale(0);visibility:hidden"
  >
    <button
      aria-label="scroll back to top"
      class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    class="home__menu"
  >
    <a
      href="/clients"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="nav-item home__nav-item"
        >
          <div
            class="nav-item__icon"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-users fa-w-20 "
              data-icon="users"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
                fill="currentColor"
              />
            </svg>
          </div>
          <span
            class="nav-item__title"
          >
            Клиенты
          </span>
        </div>
      </div>
    </a>
    <a
      href="/services"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="nav-item home__nav-item"
        >
          <div
            class="nav-item__icon"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-list fa-w-16 "
              data-icon="list"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
                fill="currentColor"
              />
            </svg>
          </div>
          <span
            class="nav-item__title"
          >
            Услуги
          </span>
        </div>
      </div>
    </a>
    <a
      href="/calendar"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="nav-item home__nav-item"
        >
          <div
            class="nav-item__icon"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-list fa-w-16 "
              data-icon="list"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
                fill="currentColor"
              />
            </svg>
          </div>
          <span
            class="nav-item__title"
          >
            Календарь
          </span>
        </div>
      </div>
    </a>
  </div>
</div>
`;
